sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfigTest.java

```diff
@@ -0,0 +1,238 @@
+    config.validate();
+  }
+
+  @Test
+  public void testPopulateDisplayData() {
+    DisplayData displayData = DisplayData.from(new HasDisplayData() {
+      @Override
+      public void populateDisplayData(DisplayData.Builder builder) {
+        config.withProjectId(PROJECT_ID)
+          .withInstanceId(INSTANCE_ID)
+          .withTableId(TABLE_ID)
+          .populateDisplayData(builder);
+      }
+    });
+
+    assertThat(displayData, hasDisplayItem(allOf(
+      hasKey("projectId"),
+      hasLabel("Bigtable Project Id"),
+      hasValue(PROJECT_ID.get()))));
+
+    assertThat(displayData, hasDisplayItem(allOf(
+      hasKey("instanceId"),
+      hasLabel("Bigtable Instance Id"),
+      hasValue(INSTANCE_ID.get()))));
+
+    assertThat(displayData, hasDisplayItem(allOf(
+      hasKey("tableId"),
+      hasLabel("Bigtable Table Id"),
+      hasValue(TABLE_ID.get()))));
+  }
+
+  @Test
+  public void testGetBigtableServiceWithDefaultService() {
+    assertEquals(SERVICE, config.withBigtableService(SERVICE).getBigtableService());
+  }
+
+  @Test
+  public void testGetBigtableServiceWithoutConfigurator() {
+    GcpOptions pipelineOptions = PipelineOptionsFactory.as(GcpOptions.class);
+
+    BigtableService service = config
+      .withProjectId(PROJECT_ID)
+      .withInstanceId(INSTANCE_ID)
+      .getBigtableService(pipelineOptions);
+
+    assertEquals(PROJECT_ID.get(), service.getBigtableOptions().getProjectId());
+    assertEquals(INSTANCE_ID.get(), service.getBigtableOptions().getInstanceId());
+    assertEquals(CredentialOptions.CredentialType.SuppliedCredentials,
+      service.getBigtableOptions().getCredentialOptions().getCredentialType());
+  }
+
+  @Test
+  public void testGetBigtableServiceWithConfigurator() {
+    SerializableFunction<BigtableOptions.Builder, BigtableOptions.Builder> configurator =
+      new SerializableFunction<BigtableOptions.Builder, BigtableOptions.Builder>() {
+        @Override
+        public BigtableOptions.Builder apply(BigtableOptions.Builder input) {
+          return input
+            .setInstanceId(INSTANCE_ID.get() + INSTANCE_ID.get())
+            .setProjectId(PROJECT_ID.get() + PROJECT_ID.get())
+            .setBulkOptions(new BulkOptions.Builder().setUseBulkApi(true).build());
+        }
+      };
+
+    GcpOptions pipelineOptions = PipelineOptionsFactory.as(GcpOptions.class);
+    pipelineOptions.setGcpCredential(new TestCredential());
+
+    BigtableService service = config
+      .withProjectId(PROJECT_ID)
+      .withInstanceId(INSTANCE_ID)
+      .withBigtableOptionsConfigurator(configurator)
+      .getBigtableService(pipelineOptions);
+
+    assertEquals(PROJECT_ID.get(), service.getBigtableOptions().getProjectId());
+    assertEquals(INSTANCE_ID.get(), service.getBigtableOptions().getInstanceId());
+    assertEquals(true, service.getBigtableOptions().getBulkOptions().useBulkApi());
+    assertEquals(CredentialOptions.CredentialType.SuppliedCredentials,
```

Comment:
This can't run on a machine that doesn't have Google application default credentials configured in their environment.

Reply:
Addressed